pkg/oc/admin/router/router.go

```diff
@@ -308,6 +311,7 @@ func NewCmdRouter(f *clientcmd.Factory, parentName, name string, out, errout io.
 	cmd.Flags().StringVar(&cfg.Ciphers, "ciphers", cfg.Ciphers, "Specifies the cipher suites to use. You can choose a predefined cipher set ('modern', 'intermediate', or 'old') or specify exact cipher suites by passing a : separated list. Not supported for F5.")
 	cmd.Flags().BoolVar(&cfg.StrictSNI, "strict-sni", cfg.StrictSNI, "Use strict-sni bind processing (do not use default cert). Not supported for F5.")
 	cmd.Flags().BoolVar(&cfg.Local, "local", cfg.Local, "If true, do not contact the apiserver")
+	cmd.Flags().IntVar(&cfg.Threads, "threads", cfg.Threads, "Specifies the amount of threads for the haproxy router.")
```

Comment:
I don't know why we would configure this - I think we want it to be autodetected by default.  Probably based on the number of cores configured for this pod.

Reply:
If there is a generally correct value we should be choosing, then we should have it autodetect.

Reply:
Auto-detecting a default sounds great, but what I wonder is if we need to also provide a way to override the default; if so, then the defaulting logic could be a followup. If not an explicit value configuration, would we at least want an overall feature flag?

Reply:
We can add the --enable-threads flag, if no --threads flag is present a default is used for the nbthreads setting.
If the --threads flag is set just check that it's value is not higher than the one in /proc/sys/kernel/threads-max

Another option could be to only have the --threads flag, if no value is set use the default but i think this could be confusing